src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -269,6 +261,7 @@ public ResponseEntity<JobModel> removeDatasetAssetSpecifications(
   public ResponseEntity<JobModel> applyDatasetDataDeletion(
       UUID id, @RequestBody @Valid DataDeletionRequest dataDeletionRequest) {
     AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    verifyDatasetAuthorization(userReq, id.toString(), IamAction.SOFT_DELETE);
```

Comment:
This seems to be new rather than a change - was this just missing? (I see a few others as well - I assume same thing)

Reply:
I think this is to make things more consistent.  We would either do the check in the request or as part of the flight.  This make it fail sooner (which I think is much better from a UX standpoint).  This is one of those cases.  So the check gets done as part of the flight here:
https://github.com/DataBiosphere/jade-data-repo/blob/5629cfb2d609222d56a133a7a55c1a45a59a94ec/src/main/java/bio/terra/service/dataset/flight/datadelete/DatasetDataDeleteFlight.java#L69

You might have noticed that Saman removed that check as part of the PR